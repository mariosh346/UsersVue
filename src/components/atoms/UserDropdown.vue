<template>
  <q-btn-dropdown
    rounded
    flat
    unelevated
  >
    <template #default>
      <q-item v-close-popup>
        <q-item-section>
          <logout-button />
        </q-item-section>
      </q-item>
    </template>
    <template #label>
      <q-avatar
        v-if="user"
        color="accent"
        alt="user photo"
      >
        <q-img
          v-if="user.photoURL"
          :src="user.photoURL"
        />
        <div
          v-else-if="userLetter"
          class="text-white"
        >
          {{ userLetter }}
        </div>
      </q-avatar>
    </template>
  </q-btn-dropdown>
</template>

<script setup lang="ts">
import { storeToRefs } from 'pinia';
import { useUserStore } from 'src/stores/userStore';
import LogoutButton from 'src/components/atoms/login/LogoutButton.vue';
import { computed } from 'vue';

const { user } = storeToRefs(useUserStore());
const userLetter = computed(() => (
  user.value?.displayName || user.value?.email
)?.[0]?.toUpperCase());
</script>
