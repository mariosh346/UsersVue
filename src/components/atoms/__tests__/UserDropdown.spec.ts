import { describe, it, expect, beforeEach, vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import UserDropdown from "../UserDropdown.vue";
import { createTestingPinia } from '@pinia/testing';

describe('UserDropdown', () => {
  let wrapper: ReturnType<typeof shallowMount>;
  const createWrapper = () => shallowMount(UserDropdown, {
    global: {
      plugins: [createTestingPinia({ createSpy: vi.fn })]
    }
  });

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('renders correctly', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
