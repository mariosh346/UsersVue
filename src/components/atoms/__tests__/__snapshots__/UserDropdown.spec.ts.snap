// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserDropdown > matches snapshot 1`] = `"<q-btn-dropdown-stub replace="false" disable="false" type="button" flat="true" outline="false" push="false" unelevated="true" square="false" rounded="true" glossy="false" fab="false" fabmini="false" nocaps="false" nowrap="false" dense="false" ripple="true" align="center" stack="false" stretch="false" transitionshow="fade" transitionhide="fade" transitionduration="300" modelvalue="false" split="false" cover="false" persistent="false" noroutedismiss="false" autoclose="false" menuanchor="bottom end" menuself="top end" disablemainbtn="false" disabledropdown="false" noiconanimation="false"></q-btn-dropdown-stub>"`;
